app/src/main/java/com/example/trackflix/fragments/update/UpdateFragment.kt

```diff
@@ -0,0 +1,184 @@
+//        super.onCreate(savedInstanceState)
+//        arguments?.let {
+//            param1 = it.getString(ARG_PARAM1)
+//            param2 = it.getString(ARG_PARAM2)
+//        }
+//    }
+
+    private val args by navArgs<UpdateFragmentArgs>()
+    private lateinit var binding: FragmentUpdateBinding
+    private lateinit var myTrackableViewModel: TrackableViewModel
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        binding = FragmentUpdateBinding.inflate(layoutInflater, container, false)
+        val view = binding.root
+
+        myTrackableViewModel = ViewModelProvider(this).get(TrackableViewModel::class.java)
+
+        binding.trackableTitle.setText(args.currentTrackable.title)
+        binding.trackableGoal.setText(args.currentTrackable.goal.toString())
+        when(args.currentTrackable.type){
+            "Book" -> binding.trackableType.check(binding.book.id)
+            "Movie" -> binding.trackableType.check(binding.movie.id)
+            "Series" -> binding.trackableType.check(binding.series.id)
+            "Game" -> binding.trackableType.check(binding.game.id)
+        }
+
+        binding.button.setOnClickListener{
+            updateTrackable()
+        }
+
+        //addDeleteOptionsMenu
+        setHasOptionsMenu(true)
+
+        return view
+    }
+
+    private fun updateTrackable() {
+        if(inputCheck()) {
+            //replace this with actual data read from our view
+            val title = binding.trackableTitle.text.toString()
+            val goal = Integer.parseInt(binding.trackableGoal.text.toString())
+            val type: String
+            when (binding.trackableType.checkedRadioButtonId) {
+                binding.book.id -> type = binding.book.text.toString()
+                binding.movie.id -> type = binding.movie.text.toString()
+                binding.series.id -> type = binding.series.text.toString()
+                binding.game.id -> type = binding.game.text.toString()
+                else -> {
+                    Toast.makeText(
+                        requireContext(),
+                        "Please fill out all fields!",
+                        Toast.LENGTH_LONG
+                    ).show()
+                    return
+                }
+            }
+            //update user
+            val updatedTrackable = Trackable(args.currentTrackable.id, title, args.currentTrackable.currentProgress, goal,type)
+            myTrackableViewModel.updateTrackable(updatedTrackable)
+            Toast.makeText(requireContext(), "Successfully updated!", Toast.LENGTH_LONG).show()
+            findNavController().navigate(R.id.action_updateFragment_to_listFragment)
+        }else{
+            Toast.makeText(requireContext(), "Please fill out all fields!", Toast.LENGTH_LONG).show()
+        }
+    }
+
+    private fun inputCheck(): Boolean {
```

Comment:
This looks very similar to the method in the create fragment. Maybe a refactoring would be good.